presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -291,6 +415,98 @@ public static double round(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(S
         return Math.floor(num * factor + 0.5) / factor;
     }
 
+    private static SqlScalarFunction decimalRoundNFunction()
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name("round")
+                .literalParameters("num_precision", "num_scale")
+                .argumentTypes("decimal(num_precision, num_scale)", StandardTypes.BIGINT)
+                .returnType("decimal(num_precision, num_scale)")
+                .build();
+        return SqlScalarFunction.builder(MathFunctions.class)
+                .signature(signature)
+                .description("remainder of given quotient")
+                .methods("roundShortDecimal", "roundLongDecimal")
+                .extraParameters(MathFunctions::decimalRoundNExtraParameters)
+                .build();
+    }
+
+    private static SqlScalarFunction decimalRoundFunction()
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name("round")
+                .literalParameters("num_precision", "num_scale")
+                .argumentTypes("decimal(num_precision, num_scale)")
+                .returnType("decimal(num_precision, num_scale)")
```

Comment:
the return type's scale should be `min(38, num_precision - num_scale + min(1, num_scale))`, we need to subtract the original scale from the precision.